newIDE/app/src/AssetStore/NewObjectFromScratch.js

```diff
@@ -0,0 +1,397 @@
+        },
+      ],
+      [translateExtensionCategory('Advanced', i18n)]: [
+        {
+          name: 'TileMap::TileMap',
+        },
+        {
+          name: 'TileMap::CollisionMask',
+        },
+        {
+          name: 'PrimitiveDrawing::Drawer',
+        },
+      ],
+    };
+    return enumeratedObjectMetadatasByCategory;
+  };
+
+  const enumeratedObjectMetadatasByCategory: {
+    [string]: Array<EnumeratedObjectMetadata>,
+  } = React.useMemo(
+    () => {
+      const enumeratedObjectMetadatasByCategory = getDefaultEnumeratedObjectMetadatasByCategory(
+        i18n
+      );
+      installedEnumeratedObjectMetadatas.forEach(
+        installedEnumeratedObjectMetadata => {
+          const category = translateExtensionCategory(
+            installedEnumeratedObjectMetadata.categoryFullName,
+            i18n
+          );
+          enumeratedObjectMetadatasByCategory[category] =
+            enumeratedObjectMetadatasByCategory[category] || [];
+          const objectIndex = enumeratedObjectMetadatasByCategory[
+            category
+          ].findIndex(
+            enumeratedObjectMetadata =>
+              enumeratedObjectMetadata.name ===
+              installedEnumeratedObjectMetadata.name
+          );
+
+          if (objectIndex !== -1) {
+            const currentEnumeratedObjectMetadata =
+              enumeratedObjectMetadatasByCategory[category][objectIndex];
+            enumeratedObjectMetadatasByCategory[category][objectIndex] = {
+              ...currentEnumeratedObjectMetadata,
+              ...installedEnumeratedObjectMetadata,
+            };
+          } else {
+            enumeratedObjectMetadatasByCategory[category] = [
+              ...enumeratedObjectMetadatasByCategory[category],
+              installedEnumeratedObjectMetadata,
+            ];
+          }
+        }
+      );
+      return enumeratedObjectMetadatasByCategory;
+    },
+    [installedEnumeratedObjectMetadatas, i18n]
+  );
+
+  const { DismissableTutorialMessage } = useDismissableTutorialMessage(
+    'intro-object-types'
+  );
+
+  return !selectedCustomObject || !selectedCustomObject.assetStorePackTag ? (
+    <ScrollView>
+      {DismissableTutorialMessage && (
+        <Line>
+          <Column expand>{DismissableTutorialMessage}</Column>
+        </Line>
+      )}
+      <List>
+        {Object.keys(enumeratedObjectMetadatasByCategory).map(category => {
+          const categoryEnumeratedObjectMetadatas =
+            enumeratedObjectMetadatasByCategory[category];
+          return (
+            <React.Fragment key={category}>
+              <Subheader>{category}</Subheader>
+              {categoryEnumeratedObjectMetadatas.map(
+                enumeratedObjectMetadata => (
+                  <ObjectListItem
+                    key={enumeratedObjectMetadata.name}
+                    enumeratedObjectMetadata={enumeratedObjectMetadata}
+                    id={`object-category-${
```

Comment:
Let's double check the onboarding still works (because it asks you to create text objects) :)

Reply:
done! all good